github4s/jvm/src/main/scala/github4s/HttpRequestBuilderExtensionJVM.scala

```diff
@@ -58,24 +64,33 @@ trait HttpRequestBuilderExtensionJVM {
                   .method(rb.httpVerb.verb)
                   .header("content-type", "application/json")
                   .asString,
-                request.url))
-          case _ ⇒ C.capture(toEntity[A](request.asString, request.url))
+                mapResponse)
+            )
+          case _ ⇒ C.capture(toEntity[A](request.asString, mapResponse))
+
         }
       }
+    }
 
+  def toEntity[A](
+      response: HttpResponse[String],
+      mapResponse: (HttpResponse[String]) => GHResponse[A]): GHResponse[A] =
+    response match {
+      case r if r.isSuccess ⇒
+        mapResponse(r)
+      case r ⇒
+        Either.left(
+          UnexpectedException(s"Failed invoking with status : ${r.code} body : \n ${r.body}"))
     }
 
-  def toEntity[A](response: HttpResponse[String], url: String)(
-      implicit D: Decoder[A]): GHResponse[A] = response match {
-    case r if r.isSuccess ⇒
-      decode[A](r.body).fold(
-        e ⇒ Either.left(JsonParsingException(e.getMessage, r.body)),
-        result ⇒ Either.right(GHResult(result, r.code, toLowerCase(r.headers)))
-      )
-    case r ⇒
-      Either.left(
-        UnexpectedException(s"Failed invoking with status : ${r.code} body : \n ${r.body}"))
-  }
+  def emptyResponse(r: HttpResponse[String]): GHResponse[Unit] =
+    Either.right(GHResult((): Unit, r.code, toLowerCase(r.headers)))
+
+  def decodeEntity[A](r: HttpResponse[String])(implicit D: Decoder[A]): GHResponse[A] =
+    decode[A](r.body).fold(
```

Comment:
Just FYI, I think you can use `bimap` (from `cats.functor.Bifunctor[Either]`) instead of `fold` and `Either.left` and `Either.right`.

Something like (untested) : 
```scala
decode[A](r.body).bimap(
  e => JsonParsingException(e.getMessage, r.body),
  result => GHResult(result, r.code, toLowerCase(r.headers))
)
```